core/src/main/scala/org/apache/spark/scheduler/TaskSetManager.scala

```diff
@@ -819,12 +816,9 @@ private[spark] class TaskSetManager(
     maybeFinishTaskSet()
   }
 
-  private[scheduler] def markPartitionCompleted(partitionId: Int, taskInfo: TaskInfo): Unit = {
+  private[scheduler] def markPartitionCompleted(partitionId: Int): Unit = {
     partitionToIndex.get(partitionId).foreach { index =>
       if (!successful(index)) {
-        if (speculationEnabled && !isZombie) {
-          successfulTaskDurations.insert(taskInfo.duration)
-        }
```

Comment:
I'm undecided about this part of the change in general (I see an argument for doing it both ways) but I'd still prefer it is not in this PR.

Reply:
then how about we use `0` as the task duration here? There is no `taskInfo` so I have to update this part.